apps/src/applab/applab.js

```diff
@@ -1250,13 +1250,17 @@ function onInterfaceModeChange(mode) {
  */
 Applab.setWorkspaceMode = function (mode) {
   var designWorkspace = document.getElementById('designWorkspace');
-  designWorkspace.style.display = (ApplabInterfaceMode.DESIGN === mode) ? 'block' : 'none';
+  if (designWorkspace) {
+    designWorkspace.style.display = (ApplabInterfaceMode.DESIGN === mode) ? 'block' : 'none';
+  }
 
   var codeWorkspaceWrapper = document.getElementById('codeWorkspaceWrapper');
   codeWorkspaceWrapper.style.display = (ApplabInterfaceMode.CODE === mode) ? 'block' : 'none';
 
   var dataWorkspaceWrapper = document.getElementById('dataWorkspaceWrapper');
-  dataWorkspaceWrapper.style.display = (ApplabInterfaceMode.DATA === mode) ? 'block' : 'none';
+  if (dataWorkspaceWrapper) {
+    dataWorkspaceWrapper.style.display = (ApplabInterfaceMode.DATA === mode) ? 'block' : 'none';
+  }
```

Comment:
For dataWorkspaceWrapper, could we not just have CodeWorkspaceContainer connect to `state.interfaceMode` and then get rid of this non React toggling of the display?

i.e.
```
connect(state => ({
  showData: state.pageConstants.hasDataMode && state.interfaceMode && state.interfaceMode === ApplabInterfaceMode.DATA,
  ...
});
```

At which point in that guys render we can just do:
```
{this.props.showData && <DataWorkspace/>
```
and don't need to render a hidden div.

I'm not sure if whether we can/can't do the same thing for designWorkspace since it's a ProtectedDiv.

Ultimately it feels a little weird to me that we have this component `CodeWorkspaceContainer` used by multiple apps, that has children (data/design workspaces) used by only one of the entry points. It's also worth noting that Gamelab has a similar concept in it's animation tab, although I think it handles it differently.